presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSplitManager.java

```diff
@@ -111,7 +121,35 @@ private static TopicPartition toTopicPartition(PartitionInfo partitionInfo)
         return new TopicPartition(partitionInfo.topic(), partitionInfo.partition());
     }
 
-    private static String readSchema(String dataSchemaLocation)
+    private Optional<String> readSchema(KafkaTableHandle tableHandle, boolean isKey)
+    {
+        Optional<String> schema = readSchemaFromSchemaRegistry(tableHandle, isKey);
+        if (schema.isPresent()) {
+            return schema;
+        }
+        if (isKey) {
+            return tableHandle.getKeyDataSchemaLocation().map(KafkaSplitManager::readSchemaFromUriOrFile);
+        }
+        else {
+            return tableHandle.getMessageDataSchemaLocation().map(KafkaSplitManager::readSchemaFromUriOrFile);
+        }
+    }
+
+    private Optional<String> readSchemaFromSchemaRegistry(KafkaTableHandle tableHandle, boolean isKey)
+    {
+        if (!isAutoResolvable(tableHandle, isKey)) {
+            return Optional.empty();
+        }
+        try {
+            SchemaMetadata schemaMetadata = schemaRegistryClient.get().getLatestSchemaMetadata(tableHandle.getTopicName() + getSuffix(isKey));
+            return Optional.of((new Schema.Parser()).parse(schemaMetadata.getSchema()).toString());
+        }
+        catch (IOException | RestClientException e) {
+            throw new TableNotFoundException(tableHandle.toSchemaTableName());
```

Comment:
IOException nor RestClientException does not mean that table does not exists, doesn't it?